lib/http.go

```diff
@@ -0,0 +1,71 @@
+package lib
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+const jsonMimeType = "application/json"
+
+// HTTPClient implements the Scheduler interface over HTTP, talking to a
+// Cron HTTPServer
+type HTTPClient struct {
+	Addr string
+}
+
+func NewHTTPClient(addr string) *HTTPClient {
+	return &HTTPClient{
+		Addr: addr,
+	}
+}
+
+func (c HTTPClient) Call(action string, ctx context.Context, params interface{}, result interface{}) error {
+	if action == "TransformMethods.Apply" {
+		log.Debugf("transform.apply start")
+		data, err := json.Marshal(params)
+		if err != nil {
+			return err
+		}
+		log.Debugf("transform.apply.data: %s", data)
+		log.Debugf("transform.apply req")
+		req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("http://%s/apply", c.Addr), bytes.NewReader(data))
+		if err != nil {
+			return err
+
+		}
+		req.Header.Set("Content-Type", jsonMimeType)
+		req.Header.Set("Accept", jsonMimeType)
+
+		log.Debugf("transform.apply res")
+		res, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+
+		log.Debugf("transform.apply body")
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
+
+		log.Debugf("transform.apply.body: %s", body)
+		resData := struct {
+			Data interface{}            `json:"data"`
+			Meta map[string]interface{} `json:"meta"`
```

Comment:
Is it the case that Meta will always have known fields? I think the api is always creating the same "envelope" containing the status code, optional error message, etc.